gadopt/utility.py

```diff
@@ -148,27 +149,15 @@ def is_continuous(expr):
 
     if isinstance(expr, ufl.indexed.Indexed):
         elem = expr.ufl_operands[0].ufl_element()
-        if isinstance(elem, ufl.MixedElement):
+        if isinstance(elem, finat.ufl.MixedElement):
             # the second operand is a MultiIndex
             assert len(expr.ufl_operands[1]) == 1
             sub_element_index, _ = elem.extract_subelement_component(int(expr.ufl_operands[1][0]))
-            elem = elem.sub_elements()[sub_element_index]
+            elem = elem.sub_elements[sub_element_index]
     else:
         elem = _get_element(expr)
 
-    family = elem.family()
-    if family == 'Lagrange' or family == 'Q':
-        return True
-    elif family == 'Discontinuous Lagrange' or family == 'DQ':
-        return False
-    elif isinstance(elem, ufl.HCurlElement) or isinstance(elem, ufl.HDivElement):
-        return False
-    elif family == 'TensorProductElement':
-        return all(is_continuous(sele) for sele in elem.sub_elements())
-    elif family == 'EnrichedElement':
-        return all(is_continuous(e) for e in elem._elements)
-    else:
-        raise NotImplementedError("Unknown finite element family")
+    return elem in ufl.H1
```

Comment:
Well that's significantly less complicated! :grin:

Reply:
Yes, well thanks for finding that - I hadn't seen this Sobolev space logic before, pretty neat indeed.